src/app/login/login.component.ts

```diff
@@ -35,11 +33,15 @@ import { RouterLink } from '@angular/router';
     },
   ],
 })
-export class LoginComponent {
-  loginForm = new FormGroup({
-    email: new FormControl('', [Validators.required, Validators.email]),
-    password: new FormControl('', [Validators.required, passwordValidator]),
-  });
```

Comment:
I think this was better, because it will create proper types for the loginForm, and its controls will not be undefined in the beginning since login FormGroup will always have a constant email and password conrol